### DISCUSSION TITLE
Import Module Federation File with Yarn Workspace

### DISCUSSION THREAD

**Discussion by LechLee** [webpack/webpack#12700] [Q&A]
Hi, is it possible to import the remoteEntry.js file using Yarn Workspace Concept

I made a sample at https://github.com/LechLee/mf-monorepo-wepback5-vue by converting it from https://github.com/jherr/mf-dog-site

The project contains two app, 
1. dog-detail
2. home (the host)

I made some changes to import the remote file using yarn workspace at https://github.com/LechLee/mf-monorepo-wepback5-vue/tree/module-federation-without-host

Here is a simple explanation on what i did:

1. Build the dog-detail app to create remoteEntry.js at dist folder
2. Import a Carousel component from remoteEntry.js at home app by **import { Carousel } from 'dog-detail/dist/remoteEntry;** at home app's App.vue

The Yarn Workspace Packages Folder:

packages
   - dog-detail
   - home

dog-detail's webpack.config.js:
```javascript
module.exports = {	
	plugins: [
		new ModuleFederationPlugin({
			name: 'dogDetail',
			filename: 'remoteEntry.js',		
			exposes: {
				'./Carousel': './src/Carousel.vue',
			}
		}),
	],
}

```

home app's App.vue:
```vue
<template>
<div class="container">
  <Carousel />
</div>
</template>

<script>
import Header from './Header.vue';
import { Carousel } from 'dog-detail/dist/remoteEntry';

export default {
  components: {
    Header,
    Carousel,
  }
}
</script>

```

However, i'm facing runtime error of **Uncaught (in promise) ReferenceError: __webpack_require__ is not defined**
![Screenshot 2021-02-17 at 8 02 06 PM](https://user-images.githubusercontent.com/10540170/108201639-13052900-715b-11eb-9925-8acea2b71ef2.png)

**Comment by alexander-akait** [comment]
Yes, you can use PnP and module federation, but I don't understand you, sorry your questions is not readable

**Reply by LechLee** [reply]
Hi @alexander-akait, i have updated the question with same code from a repo. Do hope it clears the confusion.

i tried to run 
```javascript
import { Carousel } from 'dog-detail/dist/remoteEntry';
```

But i'm facing an error of
![Screenshot 2021-02-17 at 8 02 06 PM](https://user-images.githubusercontent.com/10540170/108202842-ad19a100-715c-11eb-8ab9-c2c6e8ddef83.png)

**Reply by alexander-akait** [reply]
Something wrong with your code or configuration

**Reply by LechLee** [reply]
Hi @alexander-akait, can assist me at  https://github.com/LechLee/mf-monorepo-wepback5-vue/tree/module-federation-without-host

?

**Comment by sokra** [comment]
No you can't import the remote entry. Instead import something like `dog-detail/Carousel` and you need to add the ModuleFederationPlugin to that repo with `remotes: { "dog-detail": "..." }`

**Reply by LechLee** [reply]
Hi @sokra,
i have already done that at https://github.com/LechLee/mf-monorepo-wepback5-vue master branch
I'm exploring the possibility to achieve this without running multiple dev server during runtime as this would eat up quite a lot of processing power and memory for development purposes
